docs/SUBWORKFLOWS.md

```diff
@@ -0,0 +1,139 @@
+<MODULE1> (channel_data1)
+```
+
+To connect two modules together we need to create a channel which takes one of the outputs of the first module. To do this we use the `.out` which allows us to select the outputs of the first module and finally we specify which of the module's outputs we want :
+
+```
+channel_module2 = <MODULE1>.out.<output>
+<MODULE2> (channel_module2)
+```
+
+When a module requires multiple inputs, we don't create several channels. We create one that contains the different inputs required by the modules. To do this, you need an operator. One of the most useful is `.join`. Attention the order is important and must correspond to the desired order in the module :
+
+```
+channel_module3 = <MODULE2>.out.<output>.join(channel_data1).join(channel_data2)
+<MODULE3> (channel_module3)
+```
+
+> [!NOTE]
+> There are different types of operator depending on your needs. For a complete list, please refer to the nextflow documentation: : https://www.nextflow.io/docs/latest/operator.html#
+
+> [!WARNING]
+> The same applies to workflows, you can link modules to workflows and vice versa.
+
+At the same time, we need to edit the version files of our various modules. To do this, the first thing to do in the main is to create an empty channel:
+
+```
+ch_versions = Channel.empty()
+```
+
+Then, at the end of each module, incorporate the module version into the channel version.
+
+```
+channel_module2 = <MODULE1>.out.<output>
+<MODULE2> (channel_module2)
+ch_versions = ch_versions.mix(<MODULE2>.out.versions.first())
+```
+
+#### define your Workflow outputs.
+
+Once your `main` finish you can define the output that you want from you different modules or workflows.
+A workflow can declare one or more output channels using the `emit` keyword.
+
+```
+emit:
+    output1 = <MODULE1>.out.<output> // channel: [ val(meta), [ output ] ]
+    output2 = <MODULE2>.out.<output> // channel: [ val(meta), [ output ] ]
+```
+
+> [!NOTE]
+> As with `main`, you can create outputs containing several files using operators.
+
+Don't forget to also define the output for the version file :
+
+```
+    versions = ch_versions // channel: [ versions.yml ]
+```
+
+### Editing `./subworkflows/nf-scil/<name_of_your_workflow>/meta.yml` :
+
+Fill the sections you find relevant. There is a lot of metadata in this file, but we
+don't need to specify them all. At least define 3 `keywords`, describe the workflow'
+`inputs` and `outputs` in the order in which they appear, and add a `short description` for the use of the subworkflow.
+
+### Editing `./test/subworkflows/nf-scil/<name_of_your_workflow>/main.nf` :
+
+- define the channels of your different inputs.
+  > You can check the possibilities given by nf-core to include your dataset in the nextflow documentation : https://www.nextflow.io/docs/latest/channel.html .
+
+### Editing `./test/subworkflows/nf-scil/<name_of_your_workflow>/nextflow.config` :
+
+## Run the tests to generate the test metadata file
+
+## Last safety test
```

Comment:
You can remove this part, we decided not to press on this, we'll have to remove it from the module's doc as well.